session/interface.go

```diff
@@ -67,6 +68,10 @@ func (s State) Terminal() bool {
 // legalStateShifts is a map that defines the legal State transitions that a
 // Session can be put through.
 var legalStateShifts = map[State]map[State]bool{
+	StateReserved: {
+		StateCreated: true,
+		StateRevoked: true,
```

Comment:
Hmmm wait, I thought we were not supposed to allow a session to move directly from `StateReserved` to `StateRevoked`, as per the comment above https://github.com/lightninglabs/lightning-terminal/pull/980#discussion_r1965308238?

Imo I see no need to allow that at all, since reversed sessions get wiped on next startup of `litd`, and due to the issues described in the linked comment thread? If you change this, this should also be reflected in the `State` "flow illustration" above.

Reply:
is there a down side to just allowing this state transition? not sure it is a bad thing to allow a path to terminal state.

> The actual s.cfg.autopilot.SessionRevoked(ctx, pubKey) call in RevokeSession would not actually revoke the session on the autopilot server, as the session has not yet been registered on the server. It'll then actually get added to server, when the call to register the session to the autopilot server gets executed, after it has already been revoked on the client.

this i dont understand. Why would it get registered?

Reply:
autopilot.RevokeSession will just do nothing right?

Reply:
cool discussed offline. updating!

Reply:
Thanks for the updates :rocket:!